packages/teams-js/test/public/mail.spec.ts

```diff
@@ -0,0 +1,262 @@
+    });
+
+    it('should throw if a null itemId is supplied', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { mail: {} } });
+
+      await mail
+        .openMailItem({ itemId: null })
+        .catch(e => expect(e).toMatchObject(new Error('Must supply an itemId to openMailItem')));
+    });
+
+    it('should throw if an undefined itemId is supplied', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { mail: {} } });
+
+      await mail
+        .openMailItem({ itemId: undefined })
+        .catch(e => expect(e).toMatchObject(new Error('Must supply an itemId to openMailItem')));
+    });
+
+    it('should throw if an empty itemId is supplied', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { mail: {} } });
+
+      await mail
+        .openMailItem({ itemId: '' })
+        .catch(e => expect(e).toMatchObject(new Error('Must supply an itemId to openMailItem')));
+    });
+
+    it('should successfully throw if the openMailItem message sends and fails', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { mail: {} } });
+
+      const openMailItemPromise = mail.openMailItem(openMailItemParams);
+
+      const openMailItemMessage = utils.findMessageByFunc('mail.openMailItem');
+
+      const data = {
+        success: false,
+        error: dataError,
+      };
+
+      utils.respondToMessage(openMailItemMessage, data);
+      await openMailItemPromise.catch(e => expect(e).toMatchObject(new Error(dataError)));
+    });
+
+    it('should successfully send the openMailItem message', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { mail: {} } });
+
+      const promise = mail.openMailItem(openMailItemParams);
+
+      const openMailItemMessage = utils.findMessageByFunc('mail.openMailItem');
+
+      const data = {
+        success: true,
+        error: dataError,
+      };
+
+      utils.respondToMessage(openMailItemMessage, data);
+      await promise;
+
+      expect(openMailItemMessage).not.toBeNull();
+      expect(openMailItemMessage.args.length).toEqual(1);
+      expect(openMailItemMessage.args[0]).toStrictEqual(openMailItemParams);
+    });
+
+    it('should resolve promise after sending successful openMailItem message', async () => {
+      await utils.initializeWithContext('content');
+      utils.setRuntimeConfig({ apiVersion: 1, supports: { mail: {} } });
+
+      const promise = mail.openMailItem(openMailItemParams);
+
+      const openMailItemMessage = utils.findMessageByFunc('mail.openMailItem');
+
+      const data = {
+        success: true,
+        error: dataError,
+      };
```

Comment:
Not sure if this is something to be fixed in this PR or if it can even be fixed but it's notable that some part of the system allows `success` to be `true` but also allows there to be a non-empty `error` string. It seems like it should either be `true` or an error message, but in theory it shouldn't be allowable to have both.

Reply:
Okay, I had to look into this more and it caused me to fix a bug with the test which is good!

It turns out that "respond to message" doesn't actually expect a single data object with values, it wants separate "success" and "error message" values. The test looked like it was passing because the function was just not actually throwing (it was testing truthiness on the first argument and since it was an object it was true).

I find the the respondToMessage/communication code a little hard to follow because of just how async it is and how much it relies on passing around args objects that don't use types, but I think that the use of mysterious args is the thing I might look into more closely if I were to try to fix this. That being said, definitely outside the bounds of what this PR is doing.